### ISSUE TITLE
redundancy in rewards.rchain.coop development, admin

### ISSUE THREAD

**Issue by dckc** [rchain/bounties#872]
### Benefit to RChain

reliable rewards.rchain.coop maintenance and service even when @dckc is unavailable

See [VotingTrouble](https://github.com/rchain/bounties/wiki/VotingTrouble) for why this is particularly important / urgent.

cc @zsluedem @kitblake @jimscarver 

### Budget and Objective

*Please make the issue SMART: Specific, Measurable, Achievable, Realistic, Timely.*

_Estimated Budget of Task:_ **$[000]**
_Estimated Timeline Required to Complete the Task:_ **[days/weeks]**
_How will we measure completion?_ **one or more other admins can**

 - [ ] one or more staging environments are available to test changes (cf. #830)
 - [ ] at least two admins are authorized and able to directly administer the service

### Legal

_Task Submitter shall not submit Tasks that will involve RHOC being transacted in any manner that (i) jeopardizes RHOC’s status as a software access token or other relevant and applicable description of the RHOC as an “asset”—not a security— or (2) violates, in any manner, applicable U.S. Securities laws._

**Comment by David405** [comment]
@dckc what time can you be available for a code walkthrough? 

Also, I think some data such as voting records have been deleted from the database

**Comment by David405** [comment]
By the way, were there any recent changes you made before the application started throwing these errors?

**Comment by dckc** [comment]
Yes, there are recent changes; I detailed them in VotingTrouble.

About a code walkthrough: go ahead and suggest some times. And between now and those times, please study the code to some extent on your own and write down at least a couple questions.

**Comment by David405** [comment]
How about office hours tomorrow? I should have looked at VotingTrouble and RChain-dbr to a little more detailed extent by then.

**Comment by dckc** [comment]
Office hours tomorrow should work.

**Comment by dckc** [comment]
> Also, I think some data such as voting records have been deleted from the database

What makes you think so?

**Comment by jimscarver** [comment]
The ssh issue was solved. database management and dump loading is provides by webmin.
I think https://rewards-test.rhobot.net/ is good to go as a staging area for rchain-dbr

**Comment by David405** [comment]
@dckc, it is not an issue, at first, I couldn't find some voting records but they all seem to be in place now

**Comment by dckc** [comment]
OK... bonus points to anybody who clones rchain-dbr, makes any visible change and deploys it to rewards-test.

Also: go ahead and try loading a database dump.

**Comment by David405** [comment]
Upon further investigation, here's what I found https://stackoverflow.com/questions/15534977/mysql-cannot-add-foreign-key-constraint

**Comment by David405** [comment]
@dckc I created a [pull request](https://github.com/David405/rchain-dbr/pull/1/commits/43f66d9fabdee3c9b5601b0f8c4579034db854be) using `test` branch to commit a change to the `master` branch of the rewards app.

@jimscarver I tried to deploy the change to https://rewards-test.rhobot.net/ but I can't do that because I don't have the right permissions, kindly grant me permission so I could test changes before committing to the rewards app.

**Comment by David405** [comment]
We ( Jim and I) are working on updating the logo but have a hard time figuring out where the favicon was placed.

**Comment by dckc** [comment]
favicon? I'm not aware that there is one.

**Comment by David405** [comment]
We were able to deploy changes to the logo and favicon, see https://rewards-test.rhobot.net/

I also saw your comment on the pull request, I would edit it to accommodate further changes and resubmit it.

**Comment by David405** [comment]
@dckc I am having troubles pushing changes from my github account, I keep getting a `permission denied: 403` error, this makes it difficult to make a pull request from rewards-test to commit the changes there.

**Comment by dckc** [comment]
Did you add your own fork as a remote? You shouldn't expect to be able to
push  to someone else's fork. Getting a 403 in that case is normal.

As usual, it's best if you detail exactly what steps led to the problem.
Exactly what command? From what directory?

It's possible that you may need to clone the source to your home directory
on rewards-test. You could check the video recording of our session where
Jim pushed a change to his fork

**Comment by jimscarver** [comment]
David did a fork on github and pushed a update with and error in it.  I thought we did a git push david of the fix. If there are updates maybe another git push david is needed?
What else is needed to get david's changes into production?

**Comment by David405** [comment]
@jimscarver Yeah we did but when I tried it again to update the favicon, it threw a permission denied error

**Comment by whereyouatwimm** [comment]
I'd like to assist with this. It's my first week as a member so I'm still working on getting set up with the coop, but i've spent some of my time reviewing this issue and the source. 

@David405 wanna get together and look at your permissions error? i have feeling it's working as git is intended to and we'll need to set up your fork as remote, as Dan references above, or get your pull request merged

**Comment by David405** [comment]
Hi @whereareyouatwimm you are welcome,  we got that fixed already, thanks for showing interest. Are you on discord?

**Comment by David405** [comment]
I am available to work on this, I can remember you making the complaint while we were installing the g2 theme

**Comment by dckc** [comment]
@Viraculous asked where the theme is:

 - 469690e https://github.com/dckc/xataface-responsive-theme

(ref https://github.com/rchain/bounties/issues/792#issuecomment-412290830 )

**Comment by dckc** [comment]
@David405 I don't see any votes for budget nor reward in August. Is that on purpose?

**Comment by David405** [comment]
I haven't started entering rewards for the month yet, I usually do that at the beginning of the month

**Comment by dckc** [comment]
I encourage you to start voting earlier. Feel free to vote $1 or something, just as a place-holder or starting point for negotiation.

This month in particular, I'm travelling to RCon3, so there's no guarantee I can back your vote before the 9th.

**Comment by David405** [comment]
I have added my vote

**Comment by dckc** [comment]
@David405 it doesn't make sense to claim 100% of the budget; the objective is only 1/2 done. "at least two admins are authorized and able to directly administer the service" is still todo. Let's cut the budget in half, ok?

**Comment by David405** [comment]
@dckcI don't have a problem at all with half the budget, I only voted 100% rewards because when I saw your 30% vote I was wondering who the other 70% was for since the other two participants (yourself and @jimscarver) are covered by SOW.

Maybe I would have communicated that to you. Anyway, I have changed the budget vote to reflect the new status, probably you may need to edit yours too.

**Comment by dckc** [comment]
[rewards for this issue][1] look good now.

[1]: https://rewards.rchain.coop/index.php?-limit=30&num=872&title=&updatedAt=&state=&createdAt=&labels=&-table=issue&-action=related_records_list&-cursor=0&-skip=0&-mode=list&-recordid=issue%3Fnum%3D872&-sort=num+desc&-relationship=Rewards

**Comment by David405** [comment]
@dckc , @jimscarver and I commit to applying the "4-eyes" policy in administering services to [the rewards app](https://rewards.rchain.coop/)

**Comment by kitblake** [comment]
How much activity took place under this issue in Sept?

**Comment by dckc** [comment]
Not a lot, from a quick look at Colab discord logs. We started getting access to production for @jimscarver and @David405 but got stalled somewhere... I'm still standing by for an ssh key from @David405 , I guess.

**Comment by jimscarver** [comment]
The activity was not too successful in Sept. We did not make progress. Hopefully we can other eyes on this and sponsorship. I can do some 2nd level support but not be a primary administrator. We need to get better with git to do the job effectively.

**Comment by David405** [comment]
@dckc I wasn't aware you needed an ssh key from me, I understood from the discussion we had (@dckc, @jimscarver, me) had in the #rewards-support channel in COLab that you wanted @jimscarver and @David405 to have a single admin access.

**Comment by David405** [comment]
So what is the way forward? Perhaps we could finalize during Saturday's office hours?

**Comment by dckc** [comment]
Share your public key.

**Comment by dckc** [comment]
@whereyouatwimm thanks for agreeing to help.

As we discussed, I granted you access using an ssh public key ending in `cZGZ7wRlmK6v6lGA05v//DGUT wimm@searchingfornirvana.com`.
